src/ui/ViewModels/OrganisationViewModel.cs

```diff
@@ -27,34 +29,91 @@ public OrganisationViewModel()
+        [Required(ErrorMessage = "Enter your county")]
         public string Addr4 { get; set; }
 
-        //[Required(ErrorMessage = "Enter postcode")]
+        [Required(ErrorMessage = "Enter postcode")]
         public string Postcode { get; set; }
 
         public DateTime? LastPublishedTimestampUtc { get; set; }
 
         public EnumStatus Status { get; set; }
+
+        public static OrganisationViewModel FromEnrichmentModel(UcasInstitutionEnrichmentGetModel ucasInstitutionEnrichmentGetModel, List<TrainingProviderViewModel> aboutAccreditingTrainingProviders, UcasInstitution ucasInstitution)
+        {
+            ucasInstitutionEnrichmentGetModel = ucasInstitutionEnrichmentGetModel ?? new UcasInstitutionEnrichmentGetModel { EnrichmentModel = new InstitutionEnrichmentModel() { AccreditingProviderEnrichments = new ObservableCollection<AccreditingProviderEnrichment>() } };
+
+            var enrichmentModel = ucasInstitutionEnrichmentGetModel.EnrichmentModel;
+            ucasInstitution = ucasInstitution ?? new UcasInstitution();
+
+            var useUcasContact = 
+                string.IsNullOrWhiteSpace(enrichmentModel.Email) &&
+                string.IsNullOrWhiteSpace(enrichmentModel.Telephone) &&
+                string.IsNullOrWhiteSpace(enrichmentModel.Website) &&
+                string.IsNullOrWhiteSpace(enrichmentModel.Address1) &&
+                string.IsNullOrWhiteSpace(enrichmentModel.Address2) &&
+                string.IsNullOrWhiteSpace(enrichmentModel.Address4) &&
+                string.IsNullOrWhiteSpace(enrichmentModel.Address3) &&
+                string.IsNullOrWhiteSpace(enrichmentModel.Postcode);
+
+            var result = new OrganisationViewModel
+            {
+                InstitutionCode = ucasInstitution.InstCode,
+                InstitutionName = ucasInstitution.InstFull,
+                TrainWithUs = enrichmentModel.TrainWithUs,
+                TrainWithDisability = enrichmentModel.TrainWithDisability,
+                LastPublishedTimestampUtc = ucasInstitutionEnrichmentGetModel.LastPublishedTimestampUtc,
+                Status = ucasInstitutionEnrichmentGetModel.Status,
+                AboutTrainingProviders = aboutAccreditingTrainingProviders,
+
+                Addr1 = useUcasContact ? ucasInstitution.Addr1 : enrichmentModel.Address1,
+                Addr2 = useUcasContact ? ucasInstitution.Addr2 : enrichmentModel.Address2,
+                Addr3 = useUcasContact ? ucasInstitution.Addr3 : enrichmentModel.Address3,
+                Addr4 = useUcasContact ? ucasInstitution.Addr4 : enrichmentModel.Address4,
+                Postcode = useUcasContact ? ucasInstitution.Postcode : enrichmentModel.Postcode,
+                Url = useUcasContact ? ucasInstitution.Url : enrichmentModel.Website,
+                Telephone = useUcasContact ? ucasInstitution.Telephone : enrichmentModel.Telephone,
+                EmailAddress = useUcasContact ? ucasInstitution.Email : enrichmentModel.Email
```

Comment:
suggest moving the merge into the api so it can be shared with the publish-to-search